### PR TITLE
Add support for CHPL_LLVM_TARGETS_TO_BUILD

### PR DESCRIPTION
**Pull request by jabraham17**
Adds support for CHPL_LLVM_TARGETS_TO_BUILD, allowing users to select which backends they want to build for

Resolves https://github.com/chapel-lang/chapel/issues/26203

Tested that `CHPL_LLVM_TARGETS_TO_BUILD=host` and `CHPL_LLVM_TARGETS_TO_BUILD='host;X86'` both work

[Reviewed by @riftEmber]

### CODE REVIEW

File: third-party/llvm/Makefile [hunk 1]
```diff
@@ -12,6 +12,12 @@ else
   CHPL_LLVM_DEBUG := -DCMAKE_BUILD_TYPE=Release
 endif
 
+ifndef CHPL_LLVM_TARGETS_TO_BUILD
+  CHPL_LLVM_TARGETS_TO_BUILD := "host;X86;AArch64;NVPTX;AMDGPU"
+else
+  CHPL_LLVM_TARGETS_TO_BUILD := "$(CHPL_LLVM_TARGETS_TO_BUILD)"
```
Comment (riftEmber) [inline third-party/llvm/Makefile:18]: This just sets the variable to the same value it already had, right?

File: third-party/llvm/Makefile [hunk 2]
```diff
@@ -12,6 +12,12 @@ else
   CHPL_LLVM_DEBUG := -DCMAKE_BUILD_TYPE=Release
 endif
 
+ifndef CHPL_LLVM_TARGETS_TO_BUILD
```
Comment (riftEmber) [inline third-party/llvm/Makefile:15]: It's pretty self-explanatory, but this might still warrant a comment for posterity

**Review by riftEmber — CHANGES_REQUESTED** [review]

File: third-party/llvm/Makefile (hunk 1 above)
Comment (jabraham17) [reply third-party/llvm/Makefile:18]: Yes it does, but it also quotes them. Without it, something like `export CHPL_LLVM_TARGETS_TO_BUILD='host;X86'` doesn't work right

**Review by riftEmber — APPROVED** [review]

**Comment by bradcray** [comment]
Thanks, Jade!

**Pull request merged**